### PR TITLE
separate the DeploymentToVMDeployment to a function

### PR DESCRIPTION
**Pull request by andreaturli**
- refactor logback-test.xml
- change jsonBall to Value

### CODE REVIEW

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java [hunk 1]
```diff
@@ -90,17 +86,16 @@
...
    @Named(ComputeServiceConstants.COMPUTE_LOGGER)
    private Logger logger = Logger.NULL;
-
+   private final Json json;
    private final AzureComputeApi api;
-
    private final AzureComputeConstants azureComputeConstants;
-
    private final ProviderMetadata providerMetadata;
+   private final DeploymentToVMDeployment deploymentToVMDeployment;
```
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java]: Better declare and inject it as `Function<Deployment, VMDeployment>` to make it easier to re-bind.

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java [hunk 2]
```diff
@@ -288,7 +279,7 @@ public VMImage getImage(final String id) {
 
    @Override
    public Iterable<Location> listLocations() {
-      final Iterable<String> whiteListZoneName = findWhiteListOfRegions();
+      final Iterable<String> whiteListedRegionNames = findWhiteListOfRegions();
```
Comment (nacx) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/AzureComputeServiceAdapter.java]: You are parsing this list in several places. Just call this in the constructor and assign to a class final variable.

**Comment by nacx** [comment]
Just minors. let's merge it, since I plan to open a PR with the minor fixes in a while.

**Comment by andreaturli** [comment]
thx @nacx, merged at [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/28c56e06)

**Pull request closed without merging**